build.sbt

```diff
@@ -2,10 +2,46 @@ import Dependencies._
 
 lazy val `akkaserverless-java-sdk` = project
   .in(file("."))
-  .aggregate(sdk, testkit, tck, codegenCore, codegenJava, codegenJavaCompilationTest)
+  .aggregate(sdkCore, sdkJava, sdkScala, testkitJava, tck, codegenCore, codegenJava, codegenJavaCompilationTest)
 
-lazy val sdk = project
-  .in(file("sdk"))
+//FIXME duplicating most settings in sdkCore, sdkJava, sdkScala for now, cleanup later.
+lazy val sdkCore = project
+  .in(file("sdk/core"))
+  .enablePlugins(AkkaGrpcPlugin, BuildInfoPlugin, PublishSonatype)
+  .settings(
+    name := "akkaserverless-java-sdk-core",
+    crossPaths := false,
+    buildInfoKeys := Seq[BuildInfoKey](
+      name,
+      version,
+      "protocolMajorVersion" -> AkkaServerless.ProtocolVersionMajor,
+      "protocolMinorVersion" -> AkkaServerless.ProtocolVersionMinor,
+      "scalaVersion" -> scalaVersion.value),
+    //FIXME maybe a different package name open to suggestions.
+    buildInfoPackage := "com.akkaserverless.javasdk.impl",
+    // Generate javadocs by just including non generated Java sources
+    Compile / doc / sources := {
+      val javaSourceDir = (Compile / javaSource).value.getAbsolutePath
+      (Compile / doc / sources).value.filter(_.getAbsolutePath.startsWith(javaSourceDir))
+    },
+    // javadoc (I think java 9 onwards) refuses to compile javadocs if it can't compile the entire source path.
+    // but since we have java files depending on Scala files, we need to include ourselves on the classpath.
+    Compile / doc / dependencyClasspath := (Compile / fullClasspath).value,
+    Compile / compile / javacOptions ++= Seq("--release", "8"),
+    Compile / scalacOptions ++= Seq("-release", "8"),
+    Compile / akkaGrpcGeneratedSources := Seq(AkkaGrpc.Server),
+    Compile / akkaGrpcGeneratedLanguages := Seq(AkkaGrpc.Scala), // FIXME should be Java, but here be dragons
```

Comment:
do we need Akka gRPC in all projects/configurations?

Reply:
I'll remove this from core now, though I need to look deeper into where stuff is generated etc.

Reply:
(Since there is nothing in core yet, can always bring it back when needed)

Reply:
it looks like this is the server that the sdk provides to the proxy, so perhaps this should later move along with the implementation to core